exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -202,6 +224,16 @@ func (r *AzureManagedMachinePool) validateMaxPods() error {
 	return nil
 }
 
+func (r *AzureManagedMachinePool) validateOSDiskType() error {
+	if r.Spec.OsDiskType != "" {
+		if !(r.Spec.OsDiskType == string(containerservice.OSDiskTypeEphemeral) || r.Spec.OsDiskType == string(containerservice.OSDiskTypeManaged)) {
+			return fmt.Errorf("Allowed values of OsDiskType are %s or %s", string(containerservice.OSDiskTypeEphemeral), string(containerservice.OSDiskTypeManaged))
```

Comment:
Forgot to mention this before for maxPods validation but we should be using api-machinery field validation errors for these (similar to what we do above with field.Invalid). Error types are found here: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/field/errors.go#L93

`field.NotSupported()` seems like a good fit for this one

Reply:
Done.

Addressed the MaxPods implementation with this PR:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1920